tests/fixtures/saas/hubspot_fixtures.py

```diff
@@ -85,3 +95,79 @@ def dataset_config_hubspot(
     )
     yield dataset
     dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def hubspot_erasure_data(connection_config_hubspot, hubspot_erasure_identity_email) -> Generator:
+    """
+    Gets the current value of the resource and restores it after the test is complete.
+    Used for erasure tests.
+    """
+
+    connector = SaaSConnector(connection_config_hubspot)
+
+    # create contact
+    contacts_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.POST,
+        path=f"/crm/v3/objects/contacts",
+        json_body={
+            "properties": {
+                "company": "test company",
+                "email": hubspot_erasure_identity_email,
+                "firstname": "SomeoneFirstname",
+                "lastname": "SomeoneLastname",
+                "phone": "(123) 123-1234",
+                "website": "someone.net"
+            }
+        },
+    )
+    contacts_response = connector.create_client().send(contacts_request)
+    contacts_body = contacts_response.json()
+    contact_id = contacts_body["id"]
+
+    # no need to subscribe contact, since creating a contact auto-subscribes them
+
+    # Allows contact to be propagated in Hubspot before calling access / erasure requests
+    remaining_tries = 5
+    while _contact_exists(hubspot_erasure_identity_email, connector) is False:
+        if remaining_tries < 1:
+            raise Exception(f"Contact with contact id {contact_id} could not be added to Hubspot")
+        time.sleep(5)
+
+    yield contact_id
+
+    # delete contact
+    delete_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.DELETE,
+        path=f"/crm/v3/objects/contacts/{contact_id}",
+    )
+    connector.create_client().send(delete_request)
+
+    # verify contact is deleted
+    remaining_tries = 5
+    while _contact_exists(hubspot_erasure_identity_email, connector) is True:
+        if remaining_tries < 1:
+            raise Exception(f"Contact with contact id {contact_id} could not be deleted from Hubspot")
+        time.sleep(5)  # Ensures contact is deleted
+
+
+def _contact_exists(hubspot_erasure_identity_email: str, connector: SaaSConnector) -> bool:
+    """
+    Confirm whether contact exists by calling search api and comparing firstname str.
+   """
+    contact_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.POST,
+        path="/crm/v3/objects/contacts/search",
+        json_body={
+            "filterGroups": [{
+                "filters": [{
+                    "value": hubspot_erasure_identity_email,
+                    "propertyName": "email",
+                    "operator": "EQ"
+                }]
+            }]
+        },
+    )
+    contact_response = connector.create_client().send(contact_request)
+    contact_body = contact_response.json()
+    return bool(contact_body["results"] and contact_body["results"][0]["properties"]["firstname"] == "SomeoneFirstname")
```

Comment:
Let's use a constant to store `"SomeoneFirstname"` here, since this will cause 25sec+ wait times if the value ever deviates and may be annoying to debug